### DISCUSSION TITLE
Dynamically Add or Remove flatpickr

### DISCUSSION THREAD

**Discussion by Martin-1182** [livewire/livewire#4673] [1. Help] [answered]
Hi, I need to enter dates in the form dynamically. I have it like this but it doesn't work for me, I think there will be a problem here x-data =" datepicker ([@entangle](https://laracasts.com/@entangle) ($ attributes-> wire ('model'))) ". I can't fix it, can anyone advise me? thanks

### blade

```
<x-datepicker wire:model.lazy="mealDate.0" />

@foreach ($mealPlan as $index => $value)
	<x-datepicker wire:model.lazy="mealDate.{{ $value }}"/>
@endforeach

<x-button wire:click.prevent="add({{ $index }})" type="button">
```
### Voucher.php

```
    public $mealPlan = [];

    public $index = 1;

    public $mealDate;

public function add($index)
    {
        $this->index = $index + 1;

        array_push($this->mealPlan, $this->index);
    }

    public function remove($index)
    {
        unset($this->mealPlan[$index]);
    }
```
### datepicker.blade
```
 <div wire:ignore x-data="datepicker(@entangle($attributes->wire('model')))" class="m-0 p-0 relative">

        <input x-ref="datePicker" x-model="selectedDate" type="text" name="selected-date"
            class="mt-1 focus:ring-indigo-500 focus:border-indigo-500 block w-full shadow-sm sm:text-sm border-gray-300 rounded-md">
    </div>

@once

    <script>
        document.addEventListener('alpine:init', () => {
            Alpine.data('datepicker', (model) => ({
                selectedDate: model,
                init() {
                    flatpickr(this.$refs.datePicker, {
                        dateFormat: "Y-m-d",
                        disableMobile: "true",
                    })
                },
                reset() {
                    this.selectedDate = null;
                }
            }))
        })
    </script>

@endonce

```

**Comment by Artessan** [comment, marked as answer]
Hey if you are in alpine 3 you can use x-modelable.

#### Component
`
<input x-data="datepicker()" x-ref="datePicker" type="text" x-modelable="value" x-bind:id="$id('input')" x-model="value" placeholder="DD-MM-YYYY" {{ $attributes }} />
` 

#### alpine
`

document.addEventListener("alpine:init", () => {

    Alpine.data('datepicker', () => ({
            value: '',
            init() {
                flatpickr(this.$refs.datePicker, {
                    locale: 'es',
                    dateFormat: 'd-m-Y',
                    allowInput: true,
                }).config.onChange.push(function(selectedDates, dateStr, instance) {
                    this.value = dateStr;
                });
            }

    }));
});
`

**Reply by Martin-1182** [reply]
Thanks for the reply but this doesn't work for me.

**Reply by Artessan** [reply]
sure, no problem

**Reply by Martin-1182** [reply]
I used this solution of yours and it works great. I pulled alpine 3 and it's cool. THX

**Comment by joshhanley** [comment]
@Martin-1182 hmm maybe it is echoing out the full wire model instead of just the name. You should also put `@entangle` inside your script.

Maybe try this

```blade
 <div wire:ignore x-data="datepicker('{{ $attributes->wire('model')->value }}')" class="m-0 p-0 relative">

        <input x-ref="datePicker" x-model="selectedDate" type="text" name="selected-date"
            class="mt-1 focus:ring-indigo-500 focus:border-indigo-500 block w-full shadow-sm sm:text-sm border-gray-300 rounded-md">
    </div>

@once

    <script>
        document.addEventListener('alpine:init', () => {
            Alpine.data('datepicker', (model) => ({
                selectedDate: @entangle(model),
                init() {
                    flatpickr(this.$refs.datePicker, {
                        dateFormat: "Y-m-d",
                        disableMobile: "true",
                    })
                },
                reset() {
                    this.selectedDate = null;
                }
            }))
        })
    </script>

@endonce
```

Hope this helps!

**Reply by Martin-1182** [reply]
@joshhanley Thank you for your response. I'm trying to follow your answer but I get `Undefined constant "model" `  here  `selectedDate: @entangle (model).`

**Reply by Artessan** [reply]
that's cuz entangle works as php echo you need to pass a var try 
`
@entangle( $attributes->wire('model'))
`

**Reply by joshhanley** [reply]
@Martin-1182 ah yep sorry, forgot to change that bit, use `selectedDate: $wire.entangle(model),`

**Reply by Martin-1182** [reply]
@joshhanley `app.js:1662 Alpine Expression Error: $wire is not defined`

**Reply by Artessan** [reply]
this.$wire

**Reply by joshhanley** [reply]
@Martin-1182 yep try what @Artessan said with `this.$wire`
